### PR TITLE
AZD refactor - Provisioning and Deployment

### PR DESCRIPTION
**Pull request by hatboyzero**
# AZD refactor - Provisioning and Deployment

Refactor of the Starter ACA deployment to utilize the Azure Developer Command Line tool for a better end user experience.

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [ ]  I have included inline docs for my changes, where applicable
- [x]  I have successfully run a local build
- [x]  I have provided the required update scripts, where applicable
- [ ]  I have updated relevant docs, where applicable

### CODE REVIEW

File: deploy/starter/infra/app/acaService.bicep [hunk 1]
```diff
@@ -14,6 +14,10 @@ param appDefinition object
 param hasIngress bool = false
 param envSettings array = []
 param secretSettings array = []
+param serviceName string
+
+var formattedAppName = replace(name, '-', '')
+var truncatedAppName = substring(formattedAppName, 0, min(length(formattedAppName), 32))
```
Comment (jamesrcounts) [inline deploy/starter/infra/app/acaService.bicep:20]: We need not to truncate anymore.  Lets use `uniquestring`.

**Review by jamesrcounts — CHANGES_REQUESTED** [review]
There is a better way i think to create short names using uniquestring instead of truncation.  

Also the docs need to be updated don't they?  Do we need `Unified-Deploy.ps1` anymore. It looks completely broken now that the JSON files are gone.

File: deploy/starter/infra/app/acaService.bicep (hunk 1 above)
Comment (jamesrcounts) [reply deploy/starter/infra/app/acaService.bicep:20]: example

```
// Generate a unique token to be used in naming resources.
// Remove linter suppression after using.
#disable-next-line no-unused-vars
var resourceToken = toLower(uniqueString(subscription().id, environmentName, location, instance))
```

**Review by ciprianjichici — APPROVED** [review]

**Review by ciprianjichici — APPROVED** [review]

**Review by ciprianjichici — APPROVED** [review]

File: deploy/starter/infra/app/acaService.bicep (hunk 1 above)
Comment (hatboyzero) [reply deploy/starter/infra/app/acaService.bicep:20]: Using uniquestring() in resource suffix generation.

**Review by ciprianjichici — APPROVED** [review]

**Review by jamesrcounts — APPROVED** [review]

**Pull request merged**